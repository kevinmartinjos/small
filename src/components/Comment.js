import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import {Panel} from 'react-bootstrap';
import '../styles/Comment.css';

var PropTypes = React.PropTypes;

var Comment = React.createClass({
	propTypes: {
		content: PropTypes.string.isRequired,
		annotation: PropTypes.string
	},
	render() {
		return(
			<Panel className="Comment">
				{this.props.annotation &&
					<Panel className="annotation"><span>{this.props.annotation}</span></Panel>
				}
				<section className='content'>
					{ReactHtmlParser(this.props.content)}
				</section>
				
			</Panel>
		);
	}
});

export default Comment;
