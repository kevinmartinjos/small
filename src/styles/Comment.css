.Comment .annotation{
	font-weight: 400;
	font-size: 16px;
	font-style: normal;
	color: rgb(0, 0, 0, 0.8);
	letter-spacing: 0.01rem;
	line-height: 1.4;
	font-family: 'medium-content-serif-font,Georgia,Cambria,"Times New Roman",Times,serif';
}

.Comment .annotation{
	cursor: pointer;
}

.Comment .annotation span{
	background-color: rgba(233, 253, 240, 1) !important;
}

.Comment a{
	text-decoration: none;
}

.Comment a:hover{
	text-decoration: none;
}

.Comment .content{
	font-weight: 400;
	font-size: 18px;
	font-style: normal;
	color: rgb(0, 0, 0, 0.8);
	letter-spacing: -.004em;
	line-height: 1.58;
	font-family: 'medium-content-serif-font,Georgia,Cambria,"Times New Roman",Times,serif';	
}