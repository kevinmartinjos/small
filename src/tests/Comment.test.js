import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Comment from '../components/Comment';
import ReactTestUtils from 'react-addons-test-utils';

test('Comment should be rendered', () => {
	const component = renderer.create(
		<Comment content="I am a comment" />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});

test('Comment should render html', () => {
	let component = ReactTestUtils.renderIntoDocument(<Comment content="I am a <span>span</span>" />);
	let node = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'Comment');
	expect(node.textContent).toBe('I am a span');
});