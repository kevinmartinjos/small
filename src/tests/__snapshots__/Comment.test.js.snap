exports[`test Comment should be rendered 1`] = `
<div
  className="Comment panel panel-default"
  id={undefined}>
  <div
    className="panel-body">
    <section
      className="content">
      I am a comment
    </section>
  </div>
</div>
`;
